internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/util/FileUtil.java

```diff
@@ -0,0 +1,314 @@
+                        try (BufferedInputStream is = new BufferedInputStream(zip.getInputStream(entry));
+                             FileOutputStream fos = new FileOutputStream(destFile);
+                             BufferedOutputStream dest = new BufferedOutputStream(fos, buffer)) {
+                            while ((currentByte = is.read(data, 0, buffer)) != -1) {
+                                dest.write(data, 0, currentByte);
+                            }
+                        }
+                    }
+                }
+            }
+            return file.getParent() + File.separator + fileName.substring(0, fileName.length() - 4);
+        }
+    }
+
+    /**
+     * Zip the given folder.
+     *
+     * @param source      source directory to be zipped
+     * @param destination file of the created zip
+     * @throws ArchiveException
+     * @throws IOException
+     */
+    public static void zipFiles(String source, String destination) throws ArchiveException, IOException {
+        try (FileOutputStream fileOutputStream = new FileOutputStream(destination);
+             ArchiveOutputStream archive = new ArchiveStreamFactory().createArchiveOutputStream(
+                     ArchiveStreamFactory.ZIP, fileOutputStream)) {
+            File newFile = new File(source);
+            File[] fileList = newFile.listFiles();
+            if (fileList != null) {
+                for (File file : fileList) {
+                    ZipArchiveEntry entry = new ZipArchiveEntry(file.getPath());
+                    archive.putArchiveEntry(entry);
+                    try (BufferedInputStream input = new BufferedInputStream(new FileInputStream(file))) {
+                        IOUtils.copy(input, archive);
+                    } finally {
+                        archive.closeArchiveEntry();
+                    }
+                }
+            } else {
+                log.warn("File list that needs to be archived cannot be null.");
+            }
+            archive.finish();
+        }
+    }
+
+    /**
+     * Download the product pack from the FTP location.
+     *
+     * @param productPathInFtp path to download pack in the FTP location.
+     * @param productName      file name to download.
+     * @throws ScannerException
+     */
+    public static void downloadProduct(String productPathInFtp, String productName, File file) throws
+            IOException, SftpException, JSchException {
+        ChannelSftp sftp = goToFtpProductLocation(productPathInFtp);
+
+        downloadFromFtp(sftp.get(productName), file);
+        sftp.disconnect();
+    }
+
+    /**
+     * Go to the FTP location where the product pack is.
+     *
+     * @param productPathInFtp path to the product pack
+     * @return SFTP channel to access the FTP location
+     * @throws JSchException
+     * @throws SftpException
+     */
+    private static ChannelSftp goToFtpProductLocation(String productPathInFtp) throws JSchException, SftpException {
```

Comment:
Can we give a better name to this function. `goTo` does not explain what is the intention. What about `openFtpLocation`?